backend/src/LibExecution/ProgramTypes.fs

```diff
@@ -366,10 +366,8 @@ type TypeReference =
   | TUuid
   | TBytes
   // A named variable, eg `a` in `List<a>`, matches anything
-  | TVariable of string // replaces TAny
-  | TFn of
-    List<TypeReference> *  // CLEANUP: NEList
-    TypeReference // replaces TLambda
+  | TVariable of string
+  | TFn of NEList<TypeReference> * TypeReference
```

Comment:
I can't think of a case where we wouldnt have an NEList here.

Reply:
Agreed - the 'smallest' possible thing would be a `Unit -> Unit`

Reply:
thinking back to the lambda example, I suppose someone could type:
`let myFn (f: -> Int)`?

Reply:
That's true. On the other hand, we might clean that up in the parser too, something like `let myFn (f: _ -> Int)`